### PR TITLE
Turn on more compiler flags

### PR DESCRIPTION
**Pull request by balagez**
Enable some linting flags and fix most compiler warnings.

### CODE REVIEW

File: macrame-play/src/main/scala/macrame/play/enums/Enum.scala [hunk 1]
```diff
@@ -19,9 +19,9 @@ import scala.util.Try
  * object Color extends AsJson[Color]
  * }}}
  */
-trait AsJson[Enum] { self : EnumApi[Enum] ⇒
```
Comment (balagez) [inline macrame-play/src/main/scala/macrame/play/enums/Enum.scala:22]: Masks `lang.Enum` so all these trigger compiler warnings.

File: macrame-play/src/test/scala/macrame/play/enums/EnumTest.scala [hunk 2]
```diff
@@ -7,12 +7,11 @@ import org.scalatest.FunSuite
...
 import play.api.mvc.{ PathBindable, QueryStringBindable }
 
-import scala.math.Ordering
-
 class EnumTest extends FunSuite {
 
+   val yellowStr = "YELLOW"
+
    test("AsJson should work.") {
-      val yellowStr = "YELLOW"
```
Comment (balagez) [inline macrame-play/src/test/scala/macrame/play/enums/EnumTest.scala:15]: Triggers unreachable code compiler warning.

**Review by ghost — APPROVED** [review]
Yes! Especially regarding those Unicode arrows.

File: build.sbt [hunk 3]
```diff
@@ -8,18 +8,44 @@ inThisBuild(Seq(
...
-      "-language:postfixOps"
+      "-unchecked",                        // Show details of unchecked warnings.
+      "-deprecation",                      // Show details of deprecation warnings.
+      "-encoding", "UTF-8",                // Set correct encoding for Scaladoc.
+      "-feature",                          // Show details of feature warnings.
+      "-explaintypes",                     // Explain type errors in more detail.
+      "-Xcheckinit",                       // Wrap field accessors to throw an exception on uninitialized access.
+      "-Xlint",                            // Ensure best practices are being followed.
+      "-Xlint:adapted-args",               // Warn if an argument list is modified to match the receiver.
+      "-Xlint:constant",                   // Evaluation of a constant arithmetic expression results in an error.
```
Comment (nehaev) [inline build.sbt]: Some of these flags are not supported in 2.11:
```
-Xlint:constant
-Ywarn-extra-implicit
-Ywarn-unused:*
```
We might need either to drop 2.11 support, or to add a version switch.

**Review by nehaev — CHANGES_REQUESTED** [review]
We need to fix the build for 2.11

File: build.sbt (hunk 3 above)
Comment (nehaev) [reply build.sbt]: Version switch: https://github.com/gawkermedia/macrame/pull/8

Comment (balagez) [reply build.sbt]: We'll deal with the version number before releasing 2.13

**Review by nehaev — APPROVED** [review]

**Pull request merged**